Userland/Libraries/LibUnicode/IDNA.h

```diff
@@ -33,4 +33,36 @@ struct Mapping {
     Utf32View mapped_to;
 };
 
+enum class CheckHyphens {
+    No,
+    Yes,
+};
+
+enum class CheckBidi {
+    No,
+    Yes,
+};
+
+enum class CheckJoiners {
+    No,
+    Yes,
+};
+
+enum class UseStd3AsciiRules {
+    No,
+    Yes,
+};
+
+enum class TransitionalProcessing {
+    No,
+    Yes,
+};
+
+enum class VerifyDnsLength {
+    No,
+    Yes,
+};
+
+ErrorOr<String> to_ascii(Utf8View domain_name, CheckHyphens = CheckHyphens::Yes, CheckBidi = CheckBidi::Yes, CheckJoiners = CheckJoiners::Yes, UseStd3AsciiRules = UseStd3AsciiRules::No, TransitionalProcessing = TransitionalProcessing::No, VerifyDnsLength = VerifyDnsLength::Yes);
```

Comment:
General tip, a long list of options like these go much better in a struct. For example, see: 07e9a8f79bc8abbad87a931d54ef880bcdfbd8e1

Consider e.g. the user wants to only change the default value of `VerifyDnsLength`. They then have to pass the defaults for the 5 other options. If those defaults ever change, or if a new option is added, it's a job to find and update all callers. Whereas if they're in a struct:

```c++
struct ToAsciiOptions {
    CheckHyphens check_hyphens { CheckHyphens::Yes };
    CheckBidi check_bidi { CheckBidi::Yes };
    CheckJoiners check_joiners { CheckJoiners::Yes };
    UseStd3AsciiRules use_std3_ascii_rules { UseStd3AsciiRules::No };
    TransitionalProcessing transitional_processing { TransitionalProcessing::No };
    VerifyDnsLength verify_dns_length { VerifyDnsLength::Yes };
};
```

Then call sites that want to override a single option look *much* cleaner, and it's far easier to add new options / change defaults in the future:

```c++
IDNA::to_ascii("foo", { .verify_dns_length = IDNA::VerifyDnsLength::No });
```